src/main/java/com/dnastack/ga4gh/dataconnect/repository/QueryJobDao.java

```diff
@@ -28,9 +28,12 @@
     @SqlUpdate("UPDATE query_job SET last_activity_at = now(), finished_at = now() where id = :id")
     void setQueryFinishedAndLastActivityTime(@Bind String id);
 
-    @SqlQuery("SELECT * FROM query_job WHERE last_activity_at < :lastActivity AND finished_at IS NULL")
+    @SqlQuery("SELECT * FROM query_job WHERE (last_activity_at IS NOT NULL AND started_at IS NOT NULL) AND last_activity_at < :lastActivity AND finished_at IS NULL")
```

Comment:
Why did you need to add `started_at IS NOT NULL` ?

Reply:
AH my bad, it should be `next_page_url IS NOT NULL` since that's what we use and not `started_at`